Merge Docker instructions into main hosting on Linux section (#111)
R2Northstar/NorthstarWiki: docs/hosting-a-server-with-northstar/dedicated-server/hosting-on-linux.md

# <a name="Headless_Servers">Headless Servers</a>

## Using Docker

The preferred method for running a Northstar server in Docker is using pg9182's image which can be found in this GitHub repo: [https://github.com/pg9182/northstar-dedicated](https://github.com/pg9182/northstar-dedicated)

### Prerequisites

| Requirement | Description                                             |
| ----------- | ------------------------------------------------------- |
| Kernel      | Linux 5.3+, but 4.9+ should work                        |
| CPU         | x86\_64, at least 3 cores minimum                       |
| RAM         | 2GB (Typically peaks at 1.6GB)                          |
| Disk        | 5GB                                                     |
| Network     | Recommended at least 7-20Mbps up                        |
| Docker      | Have Docker and Docker-Compose installed on the machine |

### Installation

#### Install Docker and Docker-Compose

> If you don't already have Docker and Docker-Compose installed, you can learn how to install them [here](https://docs.docker.com/engine/install/) for your distro. Once you have them installed, you can move on to the next section.

Reviewer: ```suggestion
If you don't already have Docker and Docker-Compose installed, you can learn how to install them for your distro [from this guide](https://docs.docker.com/engine/install/). Once you have them installed, you can move on to the next section.
```

### Prep Titanfall Server files

> You will need obtain Titanfall2's game files, typically by copying over your Titanfall2 installation folder and placing it on the Linux machine. Most of these files are for SP and therefore we can delete them from the installation folder and prune the file size down to \~5GB.

Reviewer: ```suggestion
You will need obtain Titanfall2's game files, typically by copying over your Titanfall2 installation folder and placing it on the Linux machine. Most of these files are for single player and therefore we can delete them from the installation folder and prune the file size down to \~5GB.
```

> Easiest way to do this is to copy the entire Titanfall2 folder to the server and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-), or alternatively download and run [this shell script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) to download only the files necessary to run a server.

Reviewer: ```suggestion
Easiest way to do this is to copy the entire Titanfall2 folder to the server and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-) \
Alternatively download and run [this shell script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) to download only the files necessary to run a server.
```

#### Copy Titanfall Folder over to the Linux Machine

1. Access the files on your Linux machine using tools like [WinSCP](https://winscp.net/eng/download.php) or any tool that allows for SSH file transfer, or transfer files via USB drive if you can physically access your server.
2. Navigate to the folder where you want to store the files. You can put them into `~/Titanfall2` for example.
3. Copy the newly pruned Titanfall folder to the server.

> ![Game files copied to `/mnt/Titanfall` as an example](https://i.postimg.cc/15HbbzFr/image.png)

Reviewer: ```suggestion
![Game files copied to `/mnt/Titanfall` as an example](https://i.postimg.cc/15HbbzFr/image.png)
```

All images should be hosted within wiki to avoid losing them cause the hoster went down

Also, copying to `/mnt/Titanfall` is terrible advice cause `/mnt/` tends to require `root` and inside pg's docker image, Titanfall files are also mounted on `/mnt/something` IIRC which tends to really confuse people.

Reviewer 2: This is how it was when I got around to it, so blame Tony or whoever added that line.

And yeah, this is pretty bad advice, I'll probably change it to be something like `/home/Titanfall2` or just outright remove that since it doesn't seem to really be necessary? On the fence about it.

Reviewer 2: changes made & bad advice/links removed

#### Copy mods to server

> If you have configured some mods, these can be placed at a similar location, like `home/YOUR-USERNAME/mods` for example

Reviewer: ```suggestion
If you have configured some mods, these can be placed at a similar location, like `home/YOUR-USERNAME/mods` for example.

Note that Northstar itself is already installed in the Docker image. Therefore unless you are testing a development version you should NOT copy over the `Northstar.XYZ` mods.
```

If you have configured some mods,these can be placed at a similar location, like `~/mods` for example

> ![Mods placed inside /mnt/mods](https://i.postimg.cc/tRD5jnrJ/image.png)

Reviewer: again, `/mnt/` is a bad place and images should be hosted first party not external

Reviewer 2: (this is how it was when I got to iiiiit)

> We are going to be using Docker-Compose to set up our container. This gives us much more flexibility and allows us to make changes to the start up arguments much cleaner. This example will let you start your servers without needing to have your compose in the same folder and directory as Titanfall 2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.

Reviewer: ```suggestion
We are going to be using Docker-Compose to set up our container. This gives us much more flexibility and allows us to make changes to the start up arguments much cleaner. This example will let you start your servers without needing to have your compose file in the same folder and directory as Titanfall 2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.
```

   pull_policy: always 
   environment:
     - NS_PORT=37015
     - NS_PORT_AUTH=8081
     - 'NS_SERVER_NAME=Enter Server Name here'
     - 'NS_SERVER_DESC=Enter your description here'
     - NS_EXTRA_ARGUMENTS=
       +setplaylist aitdm # Attrition
       +mp_gamemode aitdm # Attrition
       +map mp_angel_city
       +ns_private_match_countdown_length 0
       +ns_should_return_to_lobby 0
       +net_compresspackets_minsize 64
       +net_compresspackets 1
       +spewlog_enable 0
       +sv_maxrate 127000
   volumes:
     - /home/YOUR_USERNAME_HERE/Titanfall2:/mnt/titanfall:ro
     - /home/YOUR_USERNAME_HERE/Titanfall2/mods:/mnt/mods:ro
   ports:
     - '37015:37015/udp'
     - '8081:8081/tcp'
   restart: always
```

A list of all the CONVARs are [here](../../hosting-a-server-with-northstar/dedicated-server#convars)

A more complex docker-compose example for hosting a server can be found [here](https://github.com/pg9182/northstar-dedicated#container), along with some extra information.

### Starting

#### Run the following command

To run this container go to the folder you saved the `docker-compose.yml` in and type

```
docker-compose up
```

> Your server should be up and running now. Check the in-game server browser or the [online server list](https://northstar-stats.frontier.tf/d/R1hKFea7k/northstar-tf-server-overview?orgId=1&refresh=30s) to see if it's online. Should your server not show up or should you encounter any other issues, check the [server troubleshooting section](https://r2northstar.gitbook.io/r2northstar-wiki/hosting-a-server-with-northstar/troubleshooting) of the wiki.

Reviewer: ```suggestion
Your server should be up and running now. Check the in-game server browser or the [online server list](https://northstar.tf/servers) to see if it's online.

Should your server not show up or should you encounter any other issues, check the [server troubleshooting section](https://r2northstar.gitbook.io/r2northstar-wiki/hosting-a-server-with-northstar/troubleshooting) of the wiki.
```

`northstar-stats.frontier.tf` does show list of servers btw :P

## Using Docker

The preferred method for running a Northstar server in Docker is using pg9182's image which can be found in this GitHub repo: [https://github.com/pg9182/northstar-dedicated](https://github.com/pg9182/northstar-dedicated)

### Prerequisites

| Requirement | Description                                             |
| ----------- | ------------------------------------------------------- |
| Kernel      | Linux 5.3+, but 4.9+ should work                        |
| CPU         | x86\_64, at least 3 cores minimum                       |
| RAM         | 2GB (Typically peaks at 1.6GB)                          |
| Disk        | 5GB                                                     |
| Network     | Recommended at least 7-20Mbps up                        |
| Docker      | Have Docker and Docker-Compose installed on the machine |

### Installation

#### Install Docker and Docker-Compose

If you don't already have Docker and Docker-Compose installed, you can learn how to install them for your distro [from this guide](https://docs.docker.com/engine/install/). Once you have them installed, you can move on to the next section.

> ### Prep Titanfall Server files

Reviewer: ```suggestion
### Prepare Titanfall2 Server files
```

> You will need obtain Titanfall2's game files, typically by copying over your Titanfall2 installation folder and placing it on the Linux machine. Most of these files are for single player and therefore we can delete them from the installation folder and prune the file size down to \~5GB.

Reviewer: ```suggestion
You will need obtain Titanfall2's game files, typically by copying over your Titanfall2 installation folder and placing it on the Linux machine. Most of these files are for singleplayer and therefore we can delete them from the installation folder and prune the file size down to \~5GB.
```

Okay this one's on me, sorry ^^"

Easiest way to do this is to copy the entire Titanfall2 folder to the server and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-) \

> Alternatively download and run [this shell script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) to download only the files necessary to run a server.

Reviewer: ```suggestion
Alternatively download and run [this shell script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) to download only the game files necessary to run a server.
```

> #### Copy Titanfall Folder over to the Linux Machine

Reviewer: ```suggestion
#### Copy Titanfall2 folder over to the Linux machine
```

1. Access the files on your Linux machine using tools like [WinSCP](https://winscp.net/eng/download.php) or any tool that allows for SSH file transfer, or transfer files via USB drive if you can physically access your server.
2. Navigate to the folder where you want to store the files. You can put them into `home/YOUR-USERNAME/Titanfall2` for example.

> 3. Copy the newly pruned Titanfall folder to the server.

Reviewer: ```suggestion
3. Copy the newly pruned Titanfall2 folder to the server.
```

#### Copy mods to server

If you have configured some mods, these can be placed at a similar location, like `home/YOUR-USERNAME/mods` for example.

Note that Northstar itself is already installed in the Docker image. Therefore unless you are testing a development version you should NOT copy over the `Northstar.XYZ` mods.

#### Create docker-compose file

> We are going to be using Docker-Compose to set up our container. This gives us much more flexibility and allows us to make changes to the start up arguments much cleaner. This example will let you start your servers without needing to have your compose file in the same folder and directory as Titanfall 2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.

Reviewer: ```suggestion
We are going to be using Docker-Compose to set up our container. This gives us much more flexibility and allows us to make changes to the start up arguments much cleaner. This example will let you start your servers without needing to have your compose file in the same folder and directory as Titanfall2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.
```

```yaml
version: '3'
services:
 northstar-attrition: 
   image: ghcr.io/pg9182/northstar-dedicated:1-tf2.0.11.0 
   pull_policy: always 
   environment:
     - NS_PORT=37015
     - NS_PORT_AUTH=8081
     - 'NS_SERVER_NAME=Enter Server Name here'
     - 'NS_SERVER_DESC=Enter your description here'
     - NS_EXTRA_ARGUMENTS=
       +setplaylist aitdm # Attrition
       +mp_gamemode aitdm # Attrition
       +map mp_angel_city
       +ns_private_match_countdown_length 0
       +ns_should_return_to_lobby 0
       +net_compresspackets_minsize 64
       +net_compresspackets 1
       +spewlog_enable 0
       +sv_maxrate 127000
   volumes:
     - /home/YOUR_USERNAME_HERE/Titanfall2:/mnt/titanfall:ro
     - /home/YOUR_USERNAME_HERE/Titanfall2/mods:/mnt/mods:ro
   ports:
     - '37015:37015/udp'
     - '8081:8081/tcp'
   restart: always
```

A list of all the CONVARs are [here](../../hosting-a-server-with-northstar/dedicated-server#convars)

A more complex docker-compose example for hosting a server can be found [here](https://github.com/pg9182/northstar-dedicated#container), along with some extra information.

### Starting

#### Run the following command

> To run this container go to the folder you saved the `docker-compose.yml` in and type

Reviewer: ```suggestion
To run this container go to the folder where you saved the `docker-compose.yml` in and type
```

```

4. Update `apt` index with `sudo apt-get update` and then install docker engine, containerd, and docker compose.
```
sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
```

5. Verify docker engine is installed by running this
```
sudo docker run hello-world
```

#### Fedora
Install the `dnf-plugins-core` package (which provides the commands to manage your DNF repositories) and set up the repository.
```
sudo dnf -y install dnf-plugins-core

 sudo dnf config-manager \
    --add-repo \
    https://download.docker.com/linux/fedora/docker-ce.repo
```

Install the latest version of docker engine, containerd, and docker compose
```
sudo dnf install docker-ce docker-ce-cli containerd.io docker-compose-plugin
```
If prompted to accept the GPG key, verify that the fingerprint matches `060A 61C5 1B55 8A7F 742B 77AA C52F EB6B 621E 9F35` (cross reference this with the [GPG key fingerprint provided by dockers docs](https://docs.docker.com/engine/install/fedora/)) , and if so, accept it.

Start docker
```
sudo systemctl start docker
```

Then verify that docker is installed correctly by running this
```
sudo docker run hello-world

> ```

Reviewer: I'd rather not add instructions how to install Docker to this wiki, primarily cause they can change over time and that will require us keeping track of them and updating them when necessary.

Also as a sidenote, why do people recommend using custom repo for Docker (on Ubuntu). It's in the default repos via `apt install docker.io`

Reviewer 2: That is true, but Docker seems to keep things pretty consistent for updates and not completely change their install process, at least from what I've seen, I only added it because it seems to be fine. As for custom repo on Ubuntu/Debian, I'm not entirely sure why either, but that's what the docs for Docker say to do and I didn't want to deviate from them.

Reviewer: I'd still remove the install guide. Just link to the official docs, otherwise it takes up a large proportion just for setting up Docker IMO

Reviewer 2: eh, fair. I'll remove the docker install but still leave the install link in that case

### Prep Titanfall Server files

> You will need obtain Titanfall's game files, typically by copying over your Titanfall installation folder and placing it on the Linux machine. Most of these files are for SP and therefore we can delete from the installation folder and prune the file size down to \~5GB.

Reviewer: ```suggestion
You will need obtain Titanfall2's game files, typically by copying over your Titanfall2 installation folder and placing it on the Linux machine. Most of these files are for SP and therefore we can delete from the installation folder and prune the file size down to \~5GB.
```

> Easiest way to do this is to copy the entire Titanfall2 folder to the server and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-), or alternatively download and run [this sh script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) made by pg9182 (note: this only downloads assets necessary to run a server and does not download a copy of the game).

Reviewer: ```suggestion
Easiest way to do this is to copy the entire Titanfall2 folder to the server and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-), or alternatively download and run [this shell script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) made by pg9182 (note: this only downloads assets necessary to run a server and does not download a copy of the game).
```

#### Copy Titanfall Folder over to the Linux Machine

> 1. Access the files on your Linux machine using tools like [WinSCP](https://winscp.net/eng/download.php)

Reviewer: Y NO filezilla anymore? ;-;

Also it reads a bit as if WinSCP is required which it really isn't anything that allows file access via SSH works. We don't want to "require" a tool but "offer" some options if one doesn't know, if you get what I mean ^^

Reviewer 2: I personally can't recommend Filezilla since the Windows installer for it attempts to bundle in PUPs, which normally wouldn't be a huge deal but not everyone pays attention to the fine print and just lets things install, so for the sake of users I removed it.

Also yeah, it does read like that, I'll edit it to make it a bit lighter on it since it was more implied it was an option and not so much the only way with Filezilla in there.

Reviewer 2: Updated with `Access the files on your Linux machine using tools like [WinSCP](https://winscp.net/eng/download.php) or any tool that allows for SSH file transfer, or transfer files via USB drive if you can physically access your server.` to make it clear its a suggestion, along with mentioning USB transfer for server hosts with physical access

Reviewer: > Filezilla since the Windows installer for it attempts to bundle in PUPs

Huh, TIL. On Windows I usually install stuff via chocolatey and as such never got prompted cause it just pulls the "clean" version ^^

2. Navigate to the folder where you want to store the files. You can put them into `~/Titanfall2` for example.
3. Copy the newly pruned Titanfall folder to the server.

![Game files copied to `/mnt/Titanfall` as an example](https://i.postimg.cc/15HbbzFr/image.png)

#### Copy mods to server

If you have configured some mods,these can be placed at a similar location, like `~/mods` for example
![Mods placed inside /mnt/mods](https://i.postimg.cc/tRD5jnrJ/image.png)

#### Create docker-compose file

> We are going to be using Docker-Compose to set up our container, this gives us much more flexibility and allows us to make changes to the start up args much cleaner. This example will let you start your servers without needing to have your compose in the same folder and directory as Titanfall 2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.

Reviewer: ```suggestion
We are going to be using Docker-Compose to set up our container, this gives us much more flexibility and allows us to make changes to the start up arguments much cleaner. This example will let you start your servers without needing to have your compose in the same folder and directory as Titanfall 2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.
```

```
nano docker-compose.yml
```

> Example docker-compose.yml

Reviewer: ```suggestion
Example `docker-compose.yml`
```

formatting

```yaml
version: '3'
services:
 northstar-skirmish: 
   image: ghcr.io/pg9182/northstar-dedicated:1-tf2.0.11.0 
   pull_policy: always 
   environment:
     - NS_PORT=37015
     - NS_PORT_AUTH=8081
     - 'NS_SERVER_NAME=Enter Server Name here'
     - 'NS_SERVER_DESC=Enter your description here'
     - NS_EXTRA_ARGUMENTS=
       +setplaylist aitdm //skirmish

>        +mp_gamemode aitdm //skirmish

Reviewer: YAML uses `#` for comments not `//` and `aitdm` is Attrition, not Skirmish :P

Reviewer 2: i forgor :skull:

Reviewer 2: fixed instances of `skirmish` and `//` being used (whorps)

>        +setplaylist aitdm #attrition
>        +mp_gamemode aitdm #attrition

Reviewer: ```suggestion
       +setplaylist aitdm # Attrition
       +mp_gamemode aitdm # Attrition
```

       +map mp_angel_city
       +ns_private_match_countdown_length 0
       +ns_should_return_to_lobby 0
       +net_compresspackets_minsize 64
       +net_compresspackets 1
       +spewlog_enable 0
       +sv_maxrate 127000
   volumes:
     - /home/user/Titanfall2:/mnt/titanfall:ro

>      - /home/user/Titanfall2/mods:/mnt/mods:ro

Reviewer: I'd change the user path to something like `YOUR_USERNAME_HERE` or even just `/path/to/Titanfal2` to avoid someone just copy-pasting the file and missing to edit these lines.

   ports:
     - '37015:37015/udp'
     - '8081:8081/tcp'
   restart: always
```

> A list of all the CONVARs are [here](../../hosting-a-server-with-northstar/dedicated-server#convars)

Reviewer: Did you check that the link still works, given that you just copy-pasted this section from the file you removed? ^^

Reviewer 2: Link still works as intended :smile:

# <a name="Headless_Servers">Headless Servers</a>

## Using Docker

The Docker image for the latest version of Northstar can be found in this GitHub repo: [https://github.com/pg9182/northstar-dedicated](https://github.com/pg9182/northstar-dedicated)

### Prerequisites

| Requirement | Description                                             |
| ----------- | ------------------------------------------------------- |
| Kernel      | Linux 5.3+, but 4.9+ should work                        |
| CPU         | x86\_64, at least 3 cores minimum                       |
| RAM         | 2GB (Typically peaks at 1.6GB)                          |
| Disk        | 5GB                                                     |
| Network     | Recommended at least 7-20Mbps up                        |
| Docker      | Have Docker and Docker-Compose installed on the machine |

### Installation

#### Install Docker and Docker-Compose

> If you don't already have Docker and Docker-Compose installed, you can learn how to install them [here](https://docs.docker.com/engine/install/) for your distro. If you do have them installed, you can move on to the next section.

Reviewer: ```suggestion
If you don't already have Docker and Docker-Compose installed, you can learn how to install them [here](https://docs.docker.com/engine/install/) for your distro. Once you have them installed, you can move on to the next section.
```

### Prep Titanfall Server files

> You will need obtain Titanfall2's game files, typically by copying over your Titanfall2 installation folder and placing it on the Linux machine. Most of these files are for SP and therefore we can delete from the installation folder and prune the file size down to \~5GB.

Reviewer: ```suggestion
You will need obtain Titanfall2's game files, typically by copying over your Titanfall2 installation folder and placing it on the Linux machine. Most of these files are for SP and therefore we can delete them from the installation folder and prune the file size down to \~5GB.
```

> Easiest way to do this is to copy the entire Titanfall2 folder to the server and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-), or alternatively download and run [this shell script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) made by pg9182 (note: this only downloads assets necessary to run a server and does not download a copy of the game).

Reviewer: ```suggestion
Easiest way to do this is to copy the entire Titanfall2 folder to the server and delete the [following files](https://github.com/pg9182/northstar-dedicated#reducing-the-size-), or alternatively download and run [this shell script](https://gist.github.com/pg9182/9a962adbfc27e93237cd14e4523c9da8) to download only the files necessary to run a server.
```

I wouldn't mention the author as to not throw pg's account under the bus, shoud anyone actually come knocking :P

#### Copy Titanfall Folder over to the Linux Machine

1. Access the files on your Linux machine using tools like [WinSCP](https://winscp.net/eng/download.php) or any tool that allows for SSH file transfer, or transfer files via USB drive if you can physically access your server.
2. Navigate to the folder where you want to store the files. You can put them into `~/Titanfall2` for example.
3. Copy the newly pruned Titanfall folder to the server.

![Game files copied to `/mnt/Titanfall` as an example](https://i.postimg.cc/15HbbzFr/image.png)

#### Copy mods to server

If you have configured some mods,these can be placed at a similar location, like `~/mods` for example
![Mods placed inside /mnt/mods](https://i.postimg.cc/tRD5jnrJ/image.png)

#### Create docker-compose file

> We are going to be using Docker-Compose to set up our container, this gives us much more flexibility and allows us to make changes to the start up arguments much cleaner. This example will let you start your servers without needing to have your compose in the same folder and directory as Titanfall 2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.

Reviewer: ```suggestion
We are going to be using Docker-Compose to set up our container. This gives us much more flexibility and allows us to make changes to the start up arguments much cleaner. This example will let you start your servers without needing to have your compose in the same folder and directory as Titanfall 2 and your mods folder. Create a compose file in your home directory, or wherever is most convenient.
```

```
nano docker-compose.yml
```

> Example `docker-compose.yml`

Reviewer: ```suggestion
Example `docker-compose.yml` \
(Note that you will have to adjust some lines to make it work on your machine)
```

# <a name="Headless_Servers">Headless Servers</a>

## Using Docker

> The Docker image for the latest version of Northstar can be found in this GitHub repo: [https://github.com/pg9182/northstar-dedicated](https://github.com/pg9182/northstar-dedicated)

Reviewer: ```suggestion
The preferred method for running a Northstar server in Docker is using pg9182's image which can be found in this GitHub repo: [https://github.com/pg9182/northstar-dedicated](https://github.com/pg9182/northstar-dedicated)
```

(Note that you will have to adjust some lines to make it work on your machine)

```yaml
version: '3'
services:
 northstar-attrition: 
   image: ghcr.io/pg9182/northstar-dedicated:1-tf2.0.11.0 
   pull_policy: always 
   environment:
     - NS_PORT=37015
     - NS_PORT_AUTH=8081
     - 'NS_SERVER_NAME=Enter Server Name here'
     - 'NS_SERVER_DESC=Enter your description here'
     - NS_EXTRA_ARGUMENTS=
       +setplaylist aitdm # Attrition
       +mp_gamemode aitdm # Attrition
       +map mp_angel_city
       +ns_private_match_countdown_length 0
       +ns_should_return_to_lobby 0
       +net_compresspackets_minsize 64
       +net_compresspackets 1
       +spewlog_enable 0
       +sv_maxrate 127000
   volumes:
     - /home/YOUR_USERNAME_HERE/Titanfall2:/mnt/titanfall:ro
     - /home/YOUR_USERNAME_HERE/Titanfall2/mods:/mnt/mods:ro
   ports:
     - '37015:37015/udp'
     - '8081:8081/tcp'
   restart: always
```

A list of all the CONVARs are [here](../../hosting-a-server-with-northstar/dedicated-server#convars)

### Starting

#### Run the following command

To run this container go to the folder you saved the `docker-compose.yml` in and type

```
docker-compose up
```

> 

Reviewer: Maybe a final line of, "your server should be up and running now. Check the in-game server browser or online server list \<link here\> to see if it's online. Should your server not show up or should you encounter any other issues, check the server troubleshooting section \<link here\>." or something like that to round of the section.

Reviewer: Also maybe a mention that pg's README contains more advanced examples like running multiple servers using a single set of Titanfall2 game files, all in a single compose file.

Reviewer 2: added both an ending statement linking to the server browser & a link to pg's complex docker-compose
